RodrigoDLopez/cloudstack
# Fix duplicate entries for vpn usage

## Description

Refactor on `UsageManagerImpl.createVPNUserEvent`. Currently, the present method creates and removes a VPN user event. So this PR abstract this method into an handle, and creates methods to create and delete VPN user events.

Additionally, the new created method prevents duplicated entries for same user, with same userId, domainId and zoneId








## Types of changes


## Screenshots (if appropriate):

## How Has This Been Tested?



**environment**
**Ubuntu 16.04**
**ACS 4.11.3.0**

**Manual test**
To test VPN.USER.ADD event I duplicate a entry on `cloud_usage.usage_event ` with: 
type = 'VPN.USER.ADD' 
and processed = 0

**Expected:** Do not create a entry on `cloud_usage.usage_vpn_user`
**Result:**
2020-04-29 21:35:02,011 DEBUG [cloud.usage.UsageManagerImpl] (Usage-Job-1:null) (logid:) We do not need to create the usage VPN user [4] assigned to account [4] because it already exists.

To test VPN.USER.REMOVE event I duplicate a entry on `cloud_usage.usage_event ` with: 
type = 'VPN.USER.REMOVE' 
and processed = 0

**Expected:** If we have more than one entry for this event, we will delete then all
**Result:**
2020-04-30 12:01:29,481 DEBUG [cloud.usage.UsageManagerImpl] (Usage-Job-1:null) (logid:) Deleting vpn user [23] assigned to account [4] domain [2] and zone [0] that was created at [Thu Apr 30 11:43:52 AMT 2020].

**Result2:**
2020-04-30 12:16:06,339 WARN  [cloud.usage.UsageManagerImpl] (Usage-Job-1:null) (logid:) No usage entry for vpn user [23] assigned to account [4] domain [2] and zone [0] was found.
